File: listar_bisiestos.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char * argv[])
{
	unsigned int primero = atoi(argv[1]);
	unsigned int ultimo = atoi(argv[2]);

	unsigned int periodo = primero;

	unsigned int bisiestos[(primero + ultimo)];

	unsigned int indice = 0;

	while (periodo <= ultimo)
	{
		if ((periodo % 4) == 0)
		{
			bisiestos[indice] = periodo;
			indice += 1;
		}
		periodo += 1;
	}

	for (unsigned int bisiesto = 0; bisiesto < indice; bisiesto++)
	{
		printf("%i\n", bisiestos[bisiesto]);
	}

	return 0;
}

File: listar_bisiestos.py
#!/usr/bin/python3

# Toma de paramtero dos años siendo el segundo mayor a primero
# Devuelve de salida una lista de años bisiestos duarnte ese periodo
import sys

# Defino parametros del comando
primero = int(sys.argv[1])

ultimo = int(sys.argv[2])
# Defino variables de control
periodo = primero

bisiestos = []

while (periodo <= ultimo):
    # Si es bisiesto lo agrego a la lis
    if ((periodo % 4) == 0):
        bisiestos.append(periodo)
    # Incremento en uno el año del perido
    periodo += 1
# Muestro los años
for bisiesto in bisiestos:
    print(bisiesto)
